### PR TITLE
[79] add merge driver, refactor

### LINKED ISSUE #79: Action to resolve `VERSION` merge conflicts on all active PRs
After every merge to `dev` all other active PRs end up with a conflicted `VERSION` (between the feature head and `dev`). To resolve this we can implement an action which:

* Is deployed on merge to `dev`
* Uses the GitHub API to iterate through all *open* PRs which are merging to `dev`
* Checkout the branch of that PR using the python `sh` package to wrap up `git`
* Do `git pull && git merge dev` (or the python `sh` equivalent)
* "Manually" resolve conflicts in `VERSION`, picking the head over `dev`, then commit the change and push
* Repeat for all *open* PRs which are merging to `dev`

### PR DESCRIPTION
**Pull request by jooh**
closes #79

I also took the opportunity to clean up calver-init a bit.

### CODE REVIEW

File: scripts/calver-init [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+    echo '################################################################' >> "$1"
+    echo "${DU_MSG}" >> "$1"
+    echo "$1" has been created or updated.
+}
+
+function AddCalVerHook(){
+    mkdir -p .githooks
+    # Don't append text if it already exists
+    _AssertFirstInit ".githooks/pre-commit"
+    _AddHeader ".githooks/pre-commit"
```
Comment (jaklinger) [inline scripts/calver-init:41]: When testing, I found that the following must have been missing (not sure why it even works without it...) to ensure that the correct shell is used. In my case, without it the precommit hook was using the wrong `echo` leading to `echo -n` printing the `-n` character (weird!) (as seen in [this issue](https://stackoverflow.com/questions/11193466/echo-n-prints-n))

```suggestion
    echo '#!/usr/bin/env bash' >> .githooks/pre-commit
    _AddHeader ".githooks/pre-commit"
```

**Review by jaklinger — CHANGES_REQUESTED** [review]
Hey - thanks for this! Just a minor addition was required to get in working on my side, a weird case of the wrong "`echo`" being used in the calver script which can be resolved by adding a shebang to the pre-commit script.

I *think* it all looks good now, I set up a dummy repo and a conflict was averted by the right strategy. It's quite hard to think of edge-cases, so let's keep an eye on it!

Regarding next steps, I guess both this repo and `ojd_daps` will require a `.gitattributes` adding or editing (`ojd_daps` has a `.gitattriutes` already) and also the `.gitconfig`.

Thanks again!

**Comment by jooh** [comment]
Thanks for feedback. If this is good to merge, I can then use the updated `calver-init` to update daps_utils and ojd_daps.

**Comment by jooh** [comment]
(or maybe better to bake the application into the same PR?)

**Comment by jooh** [comment]
Anyway, on further tests this is *not* ready to merge and those shebang insertions are creating multiple shebangs. So will work more on this tomorrow.

**Comment by jooh** [comment]
Ok, should work now. I have added the application to the repo as well, so the merge driver will be there once you run `install.sh`.

**Review by jaklinger — APPROVED** [review]
tldr: Good to merge

What I did:

* created a new test repo `test`
* used the new `calver-init`
* commented out `incrementVersion`
* Created two branches with identical conflicting changes in a file called `README` and `test/VERSION`
* On merge, there was a conflict in `README` as expected, but no conflict in `test/VERSION`, thanks to the merge driver

In short, works out-of-the-box, thanks for this
